src/vm_x86.dasc

```diff
@@ -124,16 +124,46 @@
 |.define CFRAME_SPACE,	aword*7			// Delta for esp (see <--).
 |.macro saveregs_
 |  push edi; push esi; push ebx
+|.if WIN
+|  push 0
+|  fs; push dword [0]
+|.endif
 |  sub esp, CFRAME_SPACE
 |.endmacro
 |.macro saveregs
 |  push ebp; saveregs_
 |.endmacro
 |.macro restoreregs
 |  add esp, CFRAME_SPACE
+|.if WIN
+|  fs; pop dword [0]
+|  pop edi					// Just to increment esp by 4
+|.endif
 |  pop ebx; pop esi; pop edi; pop ebp
 |.endmacro
 |
+|.if WIN
```

Comment:
AFAIK there's an MSVC flag that enables SSE (plus the stack alignment) and it's turned on by default in recent versions. Also, all game engines are probably built with this flag.

Note the consequences are wider reaching than just the LuaJIT DLL build -- it can call back into C code in the EXE or another DLL.